lib/wallet/account/hot-wallet-account.ts

```diff
@@ -34,7 +41,156 @@ export class HotWalletAccount implements IWallet {
+    const scriptUTXOs = await this.rpcClient.getUtxosByAddresses({
+      addresses: [p2SHAddress.toString()],
+    });
+
+    const scriptUtxo = scriptUTXOs.entries.find(
+      (entry) => entry.outpoint.transactionId === commitTxId,
+    );
+
+    if (!scriptUtxo) {
+      throw new Error("Could not find script UTXO");
+    }
+
+    const { transactionId: revealTxId, confirm: revealTxIdConfirm } =
+      await this.revealScript(
+        scriptBuilder,
+        scriptUtxo,
+        revealPriorityFee,
+        extraOutputs,
+        options?.waitingForReveal,
+      );
+
+    // Wait for the reveal transaction to be removed to the UTXO set of the P2SH address
+    // TODO: yield failed status and retry if timeout
+    await revealTxIdConfirm;
+
+    yield {
+      status: "completed" as const,
+      commitTxId: commitTxId,
+      revealTxId: revealTxId,
+    };
+  }
+
+  private async commitScript(p2SHAddress: string) {
+    const publicKey = this.getPublicKey();
+    const address = publicKey.toAddress(this.networkId);
+
+    // Create the commit transaction
+    const { entries } = await this.rpcClient.getUtxosByAddresses({
+      addresses: [address.toString()],
+    });
+    const { transactions: pendingTxs } = await createTransactions({
+      priorityEntries: [],
+      entries,
+      outputs: [
+        {
+          address: p2SHAddress,
+          amount: kaspaToSompi(Amount.ScriptUtxoAmount)!,
+        },
+      ],
+      priorityFee: kaspaToSompi(Fee.Base.toString())!,
+      changeAddress: address.toString(),
+      networkId: this.networkId,
+    });
+
+    const pending = pendingTxs[0];
+    const signedTx = await this.signTx(pending.transaction);
+
+    const confirm = waitTxForAddress(
+      this.rpcClient,
+      address.toString(),
+      signedTx.id,
+    );
+
+    const { transactionId } = await this.rpcClient.submitTransaction({
+      transaction: signedTx,
+    });
+
+    return {
+      transactionId,
+      confirm,
+    };
+  }
+
+  private async revealScript(
+    script: ScriptBuilder,
+    scriptUtxo: IUtxoEntry,
+    priorityFee: string,
+    extraOutputs: PaymentOutput[] = [],
+    waiting = false,
+  ) {
+    const address = this.getAddress();
+    const { entries } = await this.rpcClient.getUtxosByAddresses([address]);
+
+    const { transactions: revealPendingTxs } = await createTransactions({
+      priorityEntries: [scriptUtxo],
+      entries,
+      outputs: extraOutputs.map((output) => toKaspaPaymentOutput(output)),
+      changeAddress: address,
+      priorityFee: kaspaToSompi(priorityFee),
+      networkId: this.networkId,
+    });
+
+    // Sign the transaction with the script
+    const pendingTx = revealPendingTxs[0];
+    const signedTx = await this.signTx(pendingTx.transaction, [
+      {
+        inputIndex: 0,
+        scriptHex: script.toString(),
+      },
+    ]);
+
+    const confirm = waiting
+      ? waitTxForAddress(this.rpcClient, address, signedTx.id)
```

Comment:
`waitTxForAddress` must be before submitting transaction or the event might be lost